README.md

```diff
@@ -196,6 +198,18 @@ docker run -p 8080:8080 -it --rm mgmtapi-dse
     curl -XPOST --unix-socket /tmp/mgmtapi.sock http://localhost/api/v0/lifecycle/stop
     OK
 
+# Making changes
+
+The architecture of this repository is laid as follows, front-to-back:
+
+1. The `management-api-server/doc/openapi.json` documents the API.
+2. The server implements the HTTP verbs/endpoints under the `management-api-server/src/main/java/com/datastax/mgmtapi/resources` folder (e.g. `NodeOpsresources.java`).
+3. The server methods communicate back to the agents using `cqlService.executePreparedStatement()` calls which are routed as plaintext through a local socket. These calls return `Object` which is then serialised into JSON for return to the client.
+4. The server communicates only with the `management-api-agent-common` subproject, which holds the unversioned `CassandraAPI` interface.
+5. The `management-api-agent-common/src/main/java/com/datastax/mgmtapi/NodeOpsProvider.java` routes commands through to specific versioned instances of `CassandraAPI` which is implemented in the version 3x/4x subprojects as `CassandraAPI4x`/`CassandraAPI3x`.
+
+Any change to add endpoints or features will need to make modifications in each of the above components to ensure that they propagate through.
+
```

Comment:
Thanks for documenting this. It's all stuff you will learn the hard way when you work on it, but makes it much easier to start when it's laid out for you. Awesome!

Reply:
You're welcome! Glad you like it as I wasn't sure if I'd gotten it 100% right!